### DISCUSSION TITLE
publish timeout=1 minute

### DISCUSSION THREAD

**Discussion by AbdullahGaw** [aws/aws-iot-device-sdk-python#326] [Q&A] [answered]
I have IoT Device that publishes to MQTT sometimes the processor takes longer time to publish data and it times out, while other times it works perfectly given the same amount of data. I was wondering if there is  a way to change the publish timeout to a minute more or less.

**Comment by TwistedTwigleg** [comment, marked as answer]
I think setting [configureMQTTOperationTimeout](https://s3.amazonaws.com/aws-iot-device-sdk-python-docs/html/index.html#AWSIoTPythonSDK.MQTTLib.AWSIoTMQTTClient.configureMQTTOperationTimeout) should allow you to set the operation timeout, in seconds, to a higher value. By default, it appears to be set to 5 seconds, so calling `configureMQTTOperationTimeout(60)` should set it to 60 seconds.
